zio/detector/file.go

```diff
@@ -61,6 +62,25 @@ func OpenFile(zctx *resolver.Context, path string, cfg OpenConfig) (*zbuf.File,
 	return OpenFromNamedReadCloser(zctx, f, path, cfg)
 }
 
+type checkReader struct {
```

Comment:
With these changes, we will not be able to perform fast scanning on checkReaders. (We could; a `checkReader` could implement `scanner.ScannerAble` by calling `scanner.NewScanner` on its reader.)

I think that not allowing fast scanning matches what we're trying to accomplish with the check parameter: there's some doubt on the validity of the data source, and if there's no question about that, then you get better performance by asserting so with `-check=false`. I just wanted to raise the issue in case there were other opinions.

Reply:
I hadn't considered that ramification. (Go's type system isn't very helpful in that respect).

I agree with @alfred-landrum's  assessment but am happy to add the scanner.ScannerAble plumbing as outlined above if anyone feels like we need it.